tools/shared.py

```diff
@@ -1292,6 +1296,15 @@ def __setattr__(self, attr, value):
         logging.warning(''' - (see src/settings.js for valid values)''')
       self.attrs[attr] = value
 
+    @classmethod
+    def get(self, key):
+      return self.attrs.get(key)
+
```

Comment:
why are `get/__getitem__` needed now?

Reply:
Mostly because of `shared.JS.make_initializer` and `make_coercion`. They take `settings` as a parameter, which when `None` (or empty-dict) would use the shared Settings object. The code however uses both `settings.get(key)` and `settings[key]` syntaxes, so I just forwarded those on the Settings object so it could act more dict-y and be even more backward compatible in case there were other uses of code like that.

So I can either fix all the uses to use existing Settings APIs, or implement these dict accessors. The latter was quicker, but I can do the former if that seems cleaner.

Reply:
I see, thanks. Makes sense, lgtm as it is.

(In the long term I don't think we need the `.get()` notation (properties should always exist), but I'm not sure.)